checks.d/zk.py

```diff
@@ -96,6 +166,44 @@ def check(self, instance):
                 self.service_check('zookeeper.mode', status, message=message,
                                    tags=sc_tags)
 
+        if zk_version and LooseVersion(zk_version) > LooseVersion("3.4.0"):
+            try:
+                mntr_out = self._send_command('mntr', *cx_args)
+            except ZKConnectionFailure:
+                self.increment('zookeeper.timeouts')
+                self.report_instance_mode(hostname, 'down', tags)
+                raise
+            except:
+                e = sys.exc_info()[1]
+                print >> sys.stderr, "Error: %s" % e
+                self.increment('zookeeper.datadog_client_exception')
+                self.report_instance_mode(hostname, 'unknown', tags)
+                raise
+            else:
+                metrics, mode = self.parse_mntr(mntr_out)
+                mode_tag = "mode:%s" % mode
+                if mode != 'inactive':
+                    for name in metrics:
+                        self.gauge(name, metrics[name], tags=tags + [mode_tag])
+
+
+    def report_instance_mode(self, hostname, mode, tags):
+        tags = tags + ['mode:%s' % mode]
+        self.set('zookeeper.instances', hostname, tags=tags)
+
+        gauges = {}
+        for key in self.STATUS_TYPES:
+            gauges[key] = 0
+
+        if mode in gauges:
+            gauges[mode] = 1
+        else:
+            gauges['unknown'] = 1
+
+        for k in gauges:
+            gauge_name = 'zookeeper.instances.%s' % k
+            self.gauge(gauge_name, gauges[k])
```

Comment:
I see what you're meaning and I have yet to confirm your concern with a test.
Do note that the [current master `zk.py`](https://github.com/DataDog/dd-agent/blob/ab469ec2eed4c89daf17ad0d429a94c421185619/checks.d/zk.py#L86) does not seem to add extra host tags to the gauge calls. In the code linked above, neither `tags` or `new-tags` include a host tag. `new-tags` is set [here](https://github.com/DataDog/dd-agent/blob/ab469ec2eed4c89daf17ad0d429a94c421185619/checks.d/zk.py#L207) with no host tag. `tags` is set [here](https://github.com/DataDog/dd-agent/blob/ab469ec2eed4c89daf17ad0d429a94c421185619/checks.d/zk.py#L51) which does not specify a host tag either.

`tags` is configured from the config file and so the gauge in the update does require that additional context which I havn't added. (which I will add right away)